plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/impl/jdbc/JDBCUtils.java

```diff
@@ -1,857 +1,898 @@
+    {
+        if (e instanceof InvocationTargetException) {
+            Throwable targetException = ((InvocationTargetException) e).getTargetException();
+            if (targetException instanceof SQLException) {
+                throw (SQLException) targetException;
+            } else {
+                throw new SQLException(targetException);
+            }
+        }
+    }
+
+    @NotNull
+    public static DBPDataKind resolveDataKind(@Nullable DBPDataSource dataSource, String typeName, int typeID)
+    {
+        if (dataSource == null) {
+            return JDBCDataSource.getDataKind(typeName, typeID);
+        } else if (dataSource instanceof DBPDataTypeProvider) {
+            return ((DBPDataTypeProvider) dataSource).resolveDataKind(typeName, typeID);
+        } else {
+            return DBPDataKind.UNKNOWN;
+        }
+    }
+
+    public static String escapeWildCards(JDBCSession session, String string) {
+        if (string == null || string.isEmpty() || (string.indexOf('%') == -1 && string.indexOf('_') == -1)) {
+            return string;
+        }
+        try {
+            SQLDialect dialect = SQLUtils.getDialectFromDataSource(session.getDataSource());
+            String escapeStr = dialect.getSearchStringEscape();
+            if (CommonUtils.isEmpty(escapeStr) || escapeStr.equals(" ")) {
+                return string;
+            }
+            return string.replace("%", escapeStr + "%").replace("_", escapeStr + "_");
+        } catch (Throwable e) {
+            log.debug("Error escaping wildcard string", e);
+            return string;
+        }
+    }
+
+    public static boolean queryHasOutputParameters(SQLDialect sqlDialect, String sqlQuery) {
+        return sqlQuery.contains("?");
+    }
+
+
+    public static Long queryLong(Connection session, String sql, Object... params) throws SQLException {
+        final Number result = executeQuery(session, sql, params);
+        if (result != null) {
+            return result.longValue();
+        }
+        return null;
+    }
+
+    public static long executeInsertAutoIncrement(Connection session, String sql, Object... params) throws SQLException {
+        try (PreparedStatement dbStat = session.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS)) {
+            if (params != null) {
+                for (int i = 0; i < params.length; i++) {
+                    dbStat.setObject(i + 1, params[i]);
+                }
+            }
+            dbStat.execute();
+            return getGeneratedKey(dbStat);
+        }
+    }
+
+    public static long getGeneratedKey(PreparedStatement dbStat) throws SQLException {
+        try (final ResultSet keysRS = dbStat.getGeneratedKeys()) {
+            keysRS.next();
+            return keysRS.getLong(1);
+        }
+    }
+
+    public static <T> void fetchObjects(@NotNull Collection<? super T> objects, int limit, @NotNull JDBCSession session, @NotNull CharSequence sql,
```

Comment:
Maybe some comments for this method can be beneficial.

Reply:
Agreed.